metals/src/main/scala/scala/meta/internal/parsing/FoldingRangeExtractor.scala

```diff
@@ -159,10 +165,31 @@ final class FoldingRangeExtractor(
           Some(range(tree.pos.input, tree.pos.start + 3, tree.pos.end), true)
 
         case term: Term.Match =>
-          for {
-            token <- term.expr.findFirstTrailing(_.is[KwMatch])
-            pos = range(tree.pos.input, token.pos.end, term.pos.end)
-          } yield (pos, true)
+          val firstTwoTokens = term.expr.trailingTokens
+            .withFilter(_.isNot[Token.Whitespace])
+            .take(2)
+            .toList
+          firstTwoTokens match {
+            case List(_: Token.KwMatch, brace: Token.LeftBrace) =>
+              // with braces
+              Some(
+                range(
+                  tree.pos.input,
+                  parentDefnFoldStart.getOrElse(brace.pos.end),
+                  term.pos.end,
+                ), /* adjust */ true,
+              )
+            case (matchKw: Token.KwMatch) :: _ =>
+              // braceless
+              Some(
+                range(
+                  tree.pos.input,
+                  parentDefnFoldStart.getOrElse(matchKw.pos.end),
+                  term.pos.end,
+                ), /* adjust */ false,
+              )
+            case _ => None
```

Comment:
Could you add an example when this is needed (a code sample)

Also this might simplified and resued for catch also:
```suggestion
      
          def matchCatchBlockStart[T: ClassTag]() = term.expr.trailingTokens
            .withFilter(_.isNot[Token.Whitespace]) match {
            case  (_: T) :: (brace: Token.LeftBrace) :: _ =>
              // with braces
              Some(
                  brace.pos.end,
                ), /* adjust */ true,
              )
            case (kw: T) :: _ =>
              // braceless
              Some(
                range(
                  kw.pos.end),
                ), /* adjust */ false,
              )
            case _ => None
```

parentDefnFoldStart should not be needed if we deduplicate correctly folding ranges

Reply:
```
??? match 
{
    case ??? => ???
    case ??? => ???
    case ??? => ???
    case ??? => ???
}

??? match {
    case ??? => ???
    case ??? => ???
    case ??? => ???
    case ??? => ???
}
```
With this function detecting the position of the left brace we support both configurations and they fold like this:
```
??? match 
{...
}

??? match { ...
}
```